test/parallel/test-process-no-deprecation.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+// Flags: --expose_internals --no-warnings
+
+// The --no-warnings flag only supresses writing the warning to stderr, not the
+// emission of the corresponding event. This test file can be run without it.
+
+process.noDeprecation = true;
+
+const common = require('../common');
+const assert = require('assert');
+
+function listener(warning) {
+  common.fail('received unexpected warning');
+}
+
+process.addListener('warning', listener);
+
+const internalUtil = require('internal/util');
+internalUtil.printDeprecationMessage('Something is deprecated.');
+
+// The warning would be emitted in the next tick, so continue after that.
+process.nextTick(() => {
```

Comment:
Why not use `once` above  (line 16) then do this other stuff below from inside that handler to make sure the timing is ok?